src/main/java/com/aws/greengrass/shadowmanager/configuration/RateLimitsConfiguration.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    private static int getMaxLocalRequestRatePerThingFromTopics(Topics rateLimitsTopics) {
+        int maxLocalRequestRatePerThing = Coerce.toInt(rateLimitsTopics
+                .lookup(CONFIGURATION_MAX_LOCAL_REQUESTS_RATE_PER_THING_TOPIC)
+                .dflt(DEFAULT_LOCAL_REQUESTS_RATE));
+        Validator.validateLocalShadowRequestsPerThingPerSecond(maxLocalRequestRatePerThing);
+        return maxLocalRequestRatePerThing;
+    }
+
+    private static int getMaxOutboundUpdatesPerSecondFromTopics(Topics rateLimitsTopics) {
+        int maxOutboundUpdatesPerSecond = Coerce.toInt(rateLimitsTopics
+                .lookup(CONFIGURATION_MAX_OUTBOUND_UPDATES_PS_TOPIC)
+                .dflt(DEFAULT_MAX_OUTBOUND_SYNC_UPDATES_PS));
+        Validator.validateOutboundSyncUpdatesPerSecond(maxOutboundUpdatesPerSecond);
+        return maxOutboundUpdatesPerSecond;
+    }
+
+    private RateLimitsConfiguration(int maxLocalRequestRatePerThing,
+                                    int maxTotalLocalRequestRate,
+                                    int maxOutboundUpdatesPerSecond,
+                                    InboundRateLimiter inboundRateLimiter,
+                                    IotDataPlaneClientWrapper iotDataPlaneClientWrapper) {
+
+        this.maxLocalRequestRatePerThing = maxLocalRequestRatePerThing;
+        this.maxTotalLocalRequestRate = maxTotalLocalRequestRate;
+        this.maxOutboundUpdatesPerSecond = maxOutboundUpdatesPerSecond;
+        this.inboundRateLimiter = inboundRateLimiter;
+        this.iotDataPlaneClientWrapper = iotDataPlaneClientWrapper;
+    }
+
+    private static RateLimitsConfiguration getRateLimitsConfigurationFromTopics(Topics topics) {
+        Topics rateLimitsTopics = topics.lookupTopics(CONFIGURATION_RATE_LIMITS_TOPIC);
+        int maxTotalLocalRequestRate = getMaxTotalLocalRequestRateFromTopics(rateLimitsTopics);
+        int maxLocalRequestRatePerThing = getMaxLocalRequestRatePerThingFromTopics(rateLimitsTopics);
+        int maxOutboundUpdatesPerSecond = getMaxOutboundUpdatesPerSecondFromTopics(rateLimitsTopics);
+        InboundRateLimiter inboundRateLimiter = topics.getContext().get(InboundRateLimiter.class);
+        IotDataPlaneClientWrapper iotDataPlaneClientWrapper = topics.getContext().get(IotDataPlaneClientWrapper.class);
+
+        return new RateLimitsConfiguration(maxLocalRequestRatePerThing, maxTotalLocalRequestRate,
+                maxOutboundUpdatesPerSecond, inboundRateLimiter, iotDataPlaneClientWrapper);
+    }
+
+    /**
+     * Creates a new rate limits configuration object and triggers updates based on previous configuration.
+     *
+     * @param oldConfiguration previous configuration of the component
+     * @param serviceTopics    current configuration topics
+     * @return rate limits configuration objects
+     */
+    public static RateLimitsConfiguration from(ComponentConfiguration oldConfiguration, Topics serviceTopics) {
```

Comment:
Pass the `RateLimitsConfiguration` directly. This method does not need to depend on all the values from `ComponentConfiguration`